x/incentives/keeper/distribute_test.go

```diff
@@ -92,6 +97,164 @@ func (suite *KeeperTestSuite) TestDistribute() {
+			numPools:       1,
+			poolType:       poolmanagertypes.Concentrated,
+			gaugeCoins:     sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(5000))),
+			gaugeStartTime: defaultGaugeStartTime.Add(-5 * time.Hour),
+			expectErr:      true,
+		},
+	}
+
+	for name, tc := range tests {
+		suite.Run(name, func() {
+			// setup test
+			suite.SetupTest()
+			// We fix blocktime to ensure tests are deterministic
+			suite.Ctx = suite.Ctx.WithBlockTime(defaultGaugeStartTime)
+
+			var gauges []types.Gauge
+
+			// prepare the minting account
+			addr := sdk.AccAddress([]byte("Gauge_Creation_Addr_"))
+			// mints coins so supply exists on chain
+			suite.FundAcc(addr, coinsToMint)
+
+			// make sure the module has enough funds
+			suite.App.BankKeeper.SendCoinsFromAccountToModule(suite.Ctx, addr, types.ModuleName, coinsToMint)
+			var poolId uint64
+			// prepare a CL Pool that creates gauge at the end of createPool
+			if tc.poolType == poolmanagertypes.Concentrated {
+				for i := 0; i < tc.numPools; i++ {
+					poolId = suite.PrepareConcentratedPool().GetId()
+
+					// get the gaugeId corresponding to the CL pool
+					gaugeId, err := suite.App.PoolIncentivesKeeper.GetPoolGaugeId(suite.Ctx, poolId, currentEpoch.Duration)
+					suite.Require().NoError(err)
+
+					// get the gauge from the gaudeId
+					gauge, err := suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeId)
+					suite.Require().NoError(err)
+
+					gauge.Coins = tc.gaugeCoins
+					gauge.StartTime = tc.gaugeStartTime
+					gauges = append(gauges, *gauge)
+				}
+			}
+
+			fmt.Println(gauges)
+			// Distribute tokens from the gauge
+			totalDistributedCoins, err := suite.App.IncentivesKeeper.Distribute(suite.Ctx, gauges)
+			if tc.expectErr {
+				suite.Require().Error(err)
+
+				for _, gauge := range gauges {
+					for _, coin := range gauge.Coins {
+						// get poolId from GaugeId
+						poolId, err := suite.App.PoolIncentivesKeeper.GetPoolIdFromGaugeId(suite.Ctx, gauge.GetId(), currentEpoch.Duration)
+						suite.Require().NoError(err)
+
+						// check that incentive record wasn't created
+						_, err = suite.App.ConcentratedLiquidityKeeper.GetIncentiveRecord(suite.Ctx, poolId, coin.Denom, currentEpoch.Duration, suite.App.AccountKeeper.GetModuleAddress(types.ModuleName))
+						suite.Require().Error(err)
+					}
+				}
+			} else {
+				suite.Require().NoError(err)
+
+				// check if module amount got deducted correctly
+				balance := suite.App.BankKeeper.GetAllBalances(suite.Ctx, suite.App.AccountKeeper.GetModuleAddress(types.ModuleName))
+				actualbalanceAfterDistribution := coinsToMint.AmountOf(defaultRewardDenom).Sub(balance.AmountOf(defaultRewardDenom))
+
+				if tc.poolType == poolmanagertypes.Concentrated {
+					suite.Require().Equal(tc.expectedDistributions.AmountOf(defaultRewardDenom), actualbalanceAfterDistribution)
+				}
+
+				for _, gauge := range gauges {
+					for _, coin := range gauge.Coins {
```

Comment:
yup added a clause that checks `suite.Require().NotEqual(len(gauges), 0)`

Reply:
Also in current setup, len(gauge) ==0 will only happen if it's a balancer pool gauge, since we donot create a pool if it's anything other than type concentrated